app/src/main/java/com/wachon/spotiwrap/AppState.kt

```diff
@@ -18,12 +21,31 @@ fun rememberAppState(
 
 class AppState(val navController: NavHostController, private val context: Context) {
 
+
+    val currentRoute: String
+        @Composable get() = navController.currentBackStackEntryAsState().value?.destination?.route
+            ?: ""
+
+    fun navigatePoppingUpToStartDestination(currentRoute: String, screen: Screen) {
+        if (currentRoute != screen.route) {
+            navController.navigatePoppingUpToStartDestination(screen.route)
+        }
+    }
+
+    fun navigateAndPop(screen: Screen) {
+        navController.navigateAndPop(screen.route)
+    }
+
+    fun popBackStack() {
+        navController.navigate(Screen.Login.route)
+    }
+
     fun navigateToLogin() {
         navController.navigate(Screen.Login.route)
     }
 
     fun navigateToMenu() {
-        navController.navigate(Screen.Menu.route)
+        navController.navigate(Screen.Home.route)
```

Comment:
A rather silly suggestion, rename the method
![image](https://user-images.githubusercontent.com/38072145/236017135-52ffb581-efff-4700-b4db-1a5c0feb57a4.png)